src/multisig/multisigTransactionsOutcomeParser.ts

```diff
@@ -0,0 +1,278 @@
+            return [];
+        }
+
+        // Look for the deployment events
+        const events = transactionOnNetwork.logs.events
+            .concat(transactionOnNetwork.smartContractResults.flatMap((result) => result.logs.events))
+            .filter((event) => event.identifier === Events.MultisigDeploy);
+
+        return events.map((event) => {
+            // Assuming the contract address is in the first topic
+            const addressBytes = Buffer.from(event.topics[0]);
+            const address = Address.newFromHex(addressBytes.toString("hex"));
+            return { contractAddress: address };
+        });
+    }
+
+    /**
+     * Parses the outcome of a multisig action proposal
+     * @param transactionOnNetwork The completed transaction
+     * @returns The action ID that was created
+     */
+    parseActionProposal(transactionOnNetwork: TransactionOnNetwork): number {
+        const directCallOutcome = this.findDirectMultisigCallOutcome(transactionOnNetwork);
+
+        if (!directCallOutcome || directCallOutcome.returnCode !== "ok") {
+            throw new Err("Failed to propose action: " + directCallOutcome.returnMessage);
+        }
+
+        if (directCallOutcome.returnDataParts.length === 0) {
+            throw new Err("No action ID returned in the transaction outcome");
+        }
+
+        // Assuming the first return data part contains the action ID as bytes
+        const actionIdBytes = directCallOutcome.returnDataParts[0];
+        return parseInt(Buffer.from(actionIdBytes).toString("hex"), 16);
+    }
+
+    /**
+     * Finds the direct smart contract call outcome from a transaction
+     */
+    protected findDirectMultisigCallOutcome(transactionOnNetwork: TransactionOnNetwork): SmartContractCallOutcome {
+        let outcome = this.findDirectMultisigCallOutcomeWithinSmartContractResults(transactionOnNetwork);
+
+        if (outcome) {
+            return outcome;
+        }
+
+        outcome = this.findDirectMultisigCallOutcomeIfError(transactionOnNetwork);
+
+        if (outcome) {
+            return outcome;
+        }
+
+        outcome = this.findDirectMultisigCallOutcomeWithinWriteLogEvents(transactionOnNetwork);
+
+        if (outcome) {
+            return outcome;
+        }
+
+        return new SmartContractCallOutcome({
+            function: transactionOnNetwork.function,
+            returnCode: "",
+            returnMessage: "",
+            returnDataParts: [],
+        });
+    }
+
+    /**
+     * Similar to findDirectSmartContractCallOutcome but specifically for multisig deploy operations
+     */
+    protected findDirectMultisigDeployOutcome(transactionOnNetwork: TransactionOnNetwork): SmartContractCallOutcome {
+        return this.findDirectMultisigCallOutcome(transactionOnNetwork);
+    }
+
+    /**
+     * Finds the call outcome within smart contract results
+     */
+    protected findDirectMultisigCallOutcomeWithinSmartContractResults(
```

Comment:
Can we re-design to use the logic of:
 - https://github.com/multiversx/mx-sdk-js-core/blob/main/src/smartContracts/smartContractTransactionsOutcomeParser.ts

Without replicating the code here (perhaps we should adjust something in the generic outcome parser)?

Same idea / question for the functions below.